src/schema/v2/partner.ts

```diff
@@ -231,10 +236,15 @@ export const PartnerType = new GraphQLObjectType<any, ResolverContext>({
 
           return partnerArtworksLoader(id, gravityArgs).then(
             ({ body, headers }) => {
-              return connectionFromArraySlice(body, args, {
-                arrayLength: parseInt(headers["x-total-count"] || "0", 10),
-                sliceStart: offset,
-              })
+              const totalCount = parseInt(headers["x-total-count"] || "0", 10)
```

Comment:
I calculated `totalCount` and added it to the result twice: one for each of our two cases: the default here and the second case above when the partner is authenticated and `shallow=false`. I think it's fine but wanted to call out in case you see a way to just do it once.